### PR TITLE
Implement apply_over_axes frontend function

### LINKED ISSUE #19498: apply_over_axes

### PR DESCRIPTION
**Pull request by aazam-gh**
Close #19498

### CODE REVIEW

**Comment by aazam-gh** [comment]
There is some work to be done with for the test of this function.
In order to input ndarray I am using the expand dims fucntion in the test, but I get the following error
TypeError: expand_dims requires ndarray or scalar arguments, got <class 'ivy.data_classes.array.array.Array'> at position 0

Any tips?

**Comment by zhumakhan** [comment]
Hey, because array returnted from `dtypes_and_x` is a list of arrays. Doing something like below should fix it:
```
input_dtype, a = dtype_and_x
    
    a = a[0]
    if max(axes) >= len(a.shape):
        a = ivy.expand_dims(a, axis=0)
```
Hope this was helpful!

**Comment by zhumakhan** [comment]
Also, please remove changes made to .idea foldear and change `helpers.get_dtypes("float")` to `helpers.get_dtypes("numeric")`

**Comment by aazam-gh** [comment]
Hi thanks for the suggestions :D
I'll make the changes and test it again!

**Comment by aazam-gh** [comment]
@zhumakhan it seems the above changes still gives an error `ivy.utils.exceptions.IvyIndexError: numpy: sum: axis 1 is out of bounds for array of dimension 1`
I have tested with using the below instead to keep adjusting the dimensions
`while max(axes) >= len(a.shape):
        a = ivy.to_native(ivy.expand_dims(a, axis=len(a.shape)))`
 but i get this error instead ` TypeError: expand_dims requires ndarray or scalar arguments, got <class 'ivy.data_classes.array.array.Array'> at position 0.`

**Comment by aazam-gh** [comment]
@zaeemansari70 I had removed the changes I made to the .idea folder with the commit. Should I still revert it?

**Comment by yashjha123** [comment]
There's another alternative to generate >1-d arrays in tests. You can pass in ``min_num_dims`` to the ``helpers.dtype_and_values`` function. Also, I'd suggest you use ``helpers.dtype_values_axis`` since you also need to get axis of the array. 
Thank me later!

**Comment by aazam-gh** [comment]
@yashjha123 Thanks for the help but replacing the expand_dims function with the dtype_x_axis parameters still isn't solving the error ` TypeError: expand_dims requires ndarray or scalar arguments, got <class 'ivy.data_classes.array.array.Array'> at position 0.`

I suspect it is due to the fact that ivy.sum is returning an ivy.Array itself as it is being repeatedly applied, so the expand_dims is throwing an error. I have tried adding isInstance checks in the function as well but I think there is a better approach to it.

File: ivy/functional/frontends/jax/numpy/indexing.py [hunk 1]
```diff
@@ -80,3 +80,33 @@ def diag_indices_from(arr):
...
+            a = b
+        elif ivy.get_num_dims(b) == ivy.get_num_dims(a) - 1:
+            a = ivy.expand_dims(b, axis=axis)
+        else:
+            raise ValueError("function is not returning an array of the correct shape")
+    return a
+
+
+@to_ivy_arrays_and_back
+def apply_over_axes(func, a, axes):
```
Comment (zhumakhan) [inline ivy/functional/frontends/jax/numpy/indexing.py:99]: Same function with different implementation is added above, could you please remove one of them?

**Review by zhumakhan — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/jax/numpy/indexing.py (hunk 1 above)
Comment (aazam-gh) [reply ivy/functional/frontends/jax/numpy/indexing.py]: @zhumakhan done however still need some insight on getting the test to pass

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_indexing.py [hunk 2]
```diff
@@ -406,3 +406,44 @@ def test_jax_diag_indices_from(
...
+    dtype_x_axis,
+    on_device,
+    fn_tree,
+    frontend,
+    test_flags,
+    func,
+    axes,
+):
+    input_dtype, (a,), axis = dtype_x_axis
+    helpers.test_frontend_function(
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy/test_indexing.py:440]: This function needs `backend_to_test` keyword argument: `TypeError: test_frontend_function() missing 1 required keyword-only argument: 'backend_to_test'`

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**